python/tank/hook.py

```diff
@@ -631,17 +656,33 @@ def create_hook_instance(hook_paths, parent):
 
         3. /tmp/c.py is finally loaded in, get_hook_baseclass() now returns HookB.
 
-        4. HookC class is instantiated and method method_name is executed.
+        4. HookC class is returned.
+
+    An optional `base_class` can be provided to override the default ``Hook``
+    base class. This is useful for bundles that create hook instances at
+    execution time and wish to provide default implementation without the need
+    to configure the base hook.
 
     :param hook_paths: List of full paths to hooks, in inheritance order.
-    :param parent: Parent object. This will be accessible inside
-                   the hook as self.parent, and is typically an
-                   app, engine or core object.
+    :param base_class: A python class to use as the base class for the created
+        hook. This will override the default hook base class, ``Hook``.
     :returns: Instance of the hook.
     """
+
+    if base_class:
+        # ensure the supplied base class is a subclass of Hook
+        if not issubclass(base_class, Hook):
+            raise TankError(
+                "Error retrieving hook class. The supplied base class does not "
```

Comment:
wondering if we could tweak this to be clearer? Could we include the class name of the base class, e.g. "Invalid custom hook base class. The supplied class 'PublishBase' does not inherit from Hook" - why are we printing out the 'hook paths supplied'? it seems unrelated?

Reply:
good call. changed.